### PR TITLE
Autocompile lib Curve

### PR DESCRIPTION
**Pull request by moritz4004**
Pull request for Issue #307

Add Code to build.gradle for complie Curve25519 for exactly your OS, remove libcurve25519.so, because the file is no longer needed. 

The code only works if you make gradlew build in the qabel repo.
In qable core you get this Error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/moritz/Dokumente/foo/qabel-core/build.gradle' line: 125

* What went wrong:
A problem occurred configuring root project 'qabel-core'.
> Exception thrown while executing model rule: org.gradle.nativeplatform.plugins.NativeComponentModelPlugin$Rules#createNativeBinaries(org.gradle.platform.base.BinaryContainer, org.gradle.api.NamedDomainObjectSet<org.gradle.nativeplatform.NativeComponentSpec>, org.gradle.language.base.internal.LanguageRegistry, org.gradle.nativeplatform.toolchain.internal.NativeToolChainRegistryInternal, org.gradle.platform.base.PlatformContainer, org.gradle.nativeplatform.BuildTypeContainer, org.gradle.nativeplatform.FlavorContainer, org.gradle.internal.service.ServiceRegistry, java.io.File)
   > Could not find property 'builder' on task ':linkLinux_i386Curve25519SharedLibrary'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -27,12 +28,114 @@ dependencies {
 	compile project(':ackack')
 }
 
+task nativeHeaders {
+	String inputClass = 'de.qabel.core.crypto.Curve25519';
```
Comment (L-Henke) [inline build.gradle]: Gradle don't needs a ; at the line ending.

**Comment by L-Henke** [comment]
Maybe it would have been better to copy the script from the original source https://github.com/adammurdoch/native-platform/blob/master/build.gradle and not the already modified version. The original script can also handle OS X and FreeBSD. I further think, that unless it is absolutely impossible the qabel-core repo should remain buildable on its own.

**Comment by cburkert** [comment]
@moritz4004 You told me you're stuck here. Please describe what you tried.

BTW: Did you notice @L-Henke suggestions?

**Comment by jan-schreib** [comment]
closed in favor of #407

**Pull request closed without merging**